tests/unit/all.js

```diff
@@ -752,5 +752,56 @@ describe('code snippet example', () => {
 
       assert.equal(callCount, 1);
     });
-  })
+  });
+
+  describe('debounce', () => {
+    function debounce(func, wait, immediate) {
+      var timeout;
+      return function() {
+        var context = this, args = arguments;
+        if (immediate && !timeout) func.apply(context, args);
+        clearTimeout(timeout);
+        timeout = setTimeout(function() {
+          timeout = null;
+          if (!immediate) func.apply(context, args);
+        }, wait);
+      };
+    }
+
+    it('debouncedFn is called once per beat when immediate is false', (done) => {
+      let callCount = 0;
+      let intervalCount = 0;
+      const fn = debounce(() => callCount++, 50, false);
+
+      const beat1 = setInterval(() => {
+        if (++intervalCount >= 3) clearInterval(beat1);
+        fn();
+      }, 10);
+
+      const beat2 = setTimeout(fn, 100);
+
+      setTimeout(() => {
+        if (callCount === 2 && intervalCount === 3) done();
+        else done(`callCount = ${callCount}`);
+      }, 200);
+    });
+
+    it('debouncedFn is called once per beat when immediate is true', (done) => {
+      let callCount = 0;
+      let intervalCount = 0;
+      const fn = debounce(() => callCount++, 50, true);
+
+      const beat1 = setInterval(() => {
+        if (++intervalCount >= 3) clearInterval(beat1);
+        fn();
+      }, 10);
+
+      const beat2 = setTimeout(fn, 100);
+
+      setTimeout(() => {
+        if (callCount === 2 && intervalCount === 3) done();
+        else done(`callCount = ${callCount}`);
```

Comment:
Where is assertion? Honestly i don't understand this test, you should stub `() => callCount++` and check how many calls debounce called this method

Reply:
Actually it's my first test unit, but I think it does work. It's true I didn't write any "assert" keyword to do the assertion, but I did use ["done" which I think did the job well](https://mochajs.org/#asynchronous-code).  Although, stub may be a good way to finish the job but I think stub is not available in mochaJs or assert module. It's not wise to import extra modules when the job can be done easily, at least for me.

Reply:
@ODudek Any updates?